### PR TITLE
Add linux sll support

### PR DESCRIPTION
**Pull request by RabadanDotDev**
Work mentioned in #97. I implemented the part of the public interface I was most interested in (SlicedPacket::from_linux_sll) and all the additions/extensions to the rest of the code it needed. There are some parts missing, but if you are interested in merging, I can work on them when I get the time.

Specifically, some points missing are:
- [ ] Implementation of PacketHeaders::from_linux_sll()
- [ ] Implementation of LaxSlicedPacket::from_linux_sll()
- [ ] Implementation of LaxPacketHeaders::from_linux_sll()
- [ ] Revising the added test code, making sure it makes sense and the relevant cases are covered

### CODE REVIEW

**Comment by JulianSchmid** [comment]
Wow thanks for that PR. I will have a deep look into it tomorrow, but so far it looks incredible.

**Review by JulianSchmid — APPROVED** [review]
I don't know what to say, other than this is excellent work. Thanks a lot.

I would merge your PR and move the missing topics you mentioned in the PR description into a ticket. If you find time to implement them a new PR would be very welcome.

File: etherparse/src/link/linux_sll_header.rs [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    pub protocol_type: LinuxSllProtocolType,
+}
+
+impl LinuxSllHeader {
+    /// Serialized size of an SLL header in bytes/octets.
+    pub const LEN: usize = 16;
+
+    /// Read an SLL header from a slice and return the header & unused parts of the slice.
+    #[inline]
+    pub fn from_slice(slice: &[u8]) -> Result<(LinuxSllHeader, &[u8]), err::ReadError> {
```
Comment (JulianSchmid) [inline etherparse/src/link/linux_sll_header.rs:27]:
```suggestion
    pub fn from_slice(slice: &[u8]) -> Result<(LinuxSllHeader, &[u8]), err::linux_sll::HeaderSliceError> {
```

File: etherparse/src/link/linux_sll_header.rs [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+    }
+
+    /// Read an SLL header from a static sized byte array.
+    #[inline]
+    pub fn from_bytes(bytes: [u8; 16]) -> Result<LinuxSllHeader, err::ReadError> {
+        let packet_type = LinuxSllPacketType::try_from(u16::from_be_bytes([bytes[0], bytes[1]]))?;
+        let arp_hrd_type = ArpHardwareId::from(u16::from_be_bytes([bytes[2], bytes[3]]));
+        let sender_address_valid_length = u16::from_be_bytes([bytes[4], bytes[5]]);
+        let sender_address = [bytes[6], bytes[7], bytes[8], bytes[9], bytes[10], bytes[11], bytes[12], bytes[13]];
+        let protocol_type = LinuxSllProtocolType::try_from((arp_hrd_type, u16::from_be_bytes([bytes[14], bytes[15]])))?;
```
Comment (JulianSchmid) [inline etherparse/src/link/linux_sll_header.rs:41]:
```suggestion
        use err::linux_sll::HeaderSliceError::Content;
        
        let packet_type = LinuxSllPacketType::try_from(u16::from_be_bytes([bytes[0], bytes[1]])).map_err(Content)?;
        let arp_hrd_type = ArpHardwareId::from(u16::from_be_bytes([bytes[2], bytes[3]]));
        let sender_address_valid_length = u16::from_be_bytes([bytes[4], bytes[5]]);
        let sender_address = [bytes[6], bytes[7], bytes[8], bytes[9], bytes[10], bytes[11], bytes[12], bytes[13]];
        let protocol_type = LinuxSllProtocolType::try_from((arp_hrd_type, u16::from_be_bytes([bytes[14], bytes[15]]))).map_err(Content)?;
```

**Review by JulianSchmid — COMMENTED** [review]
Minor fixes for non-std support

**Comment by JulianSchmid** [comment]
I will fix the errors that my no-std fix introduced in the evening.

**Comment by RabadanDotDev** [comment]
> I would merge your PR and move the missing topics you mentioned in the PR description into a ticket. If you find time to implement them a new PR would be very welcome.

Ok perfect, sounds good! When I finish other work I have I'll to look into to finish implementing the rest.

**Comment by GyulyVGC** [comment]
That's an amazing addition.

Thanks for the work @RabadanDotDev and @JulianSchmid for reviewing ❤️

**Pull request merged**